### PR TITLE
add method comparison in training script

### PR DESCRIPTION
**Pull request by SCiarella**
This PR address #63 by testing which loss function approach we should use between Lux and Optim.jl.

A visual comparison:
plot_153

Observations:
* Lux is ~25% faster 
* Lux is slightly more accurate, however notice that both are **bad** so further testing of our overall a-posteriori approach is required (maybe this cnn is too weak?)
* Optim can use fancy optimization algorithms like CMA or LBFGS

Conclusions:
* Training with Lux should be our main route
* We can test fancy optimization algorithms with Optim if we put this loss in the package
* Overall, we have no example where the a-posteriori training shows any sensible improvement over random model weights

### CODE REVIEW

File: simulations/NavierStokes_2D/scripts/train_posteriori.jl [hunk 1]
```diff
@@ -45,33 +47,75 @@ function loss_posteriori(model, p, st, data)
...
+    nepochs = NEPOCHS, ad_type = Optimization.AutoZygote(),
+    alg = OptimizationOptimisers.Adam(LR), cpu = true, callback = callback)
 
+loss, tstate = lux_result
 # the trained params are:
-θ_posteriori = tstate.parameters
+θ_posteriori_lux = tstate.parameters
+
+# Function to test the results
+function validate_results(p, dataloader, nuse = 100)
```
Comment (luisaforozco) [inline simulations/NavierStokes_2D/scripts/train_posteriori.jl:85]: I am wondering if this validation an plotting should be in this file or in `evaluate_posteriori.jl`, have you thought about that?

Comment (SCiarella) [reply simulations/NavierStokes_2D/scripts/train_posteriori.jl:85]: Would be nice, however for the comparison we need to access the full training routine, while the idea of `evaluate_posteriori` is to just load a trained model. 

I think that this comparison should go to the [explorations](https://github.com/DEEPDIP-project/explorations) repo, but we include both loss functions to `src` and we show only the Lux one in `train_posteriori`. What do you think?

Comment (luisaforozco) [reply simulations/NavierStokes_2D/scripts/train_posteriori.jl:85]: Yes, I like the idea:
1. copy the script as it is to explorations (still with the loss defined in it for clarity). 
2. adding the code to `src/loss/loss_posteriori.jl`.

**Comment by luisaforozco** [comment]
I'm ok with this! I'm merging it.
